packages/app-mobile/components/get-responsive-value.ts

```diff
@@ -0,0 +1,83 @@
+// getResponsiveValue() returns the corresponding value for
+// of a particular device screen width based on a valueMap argument
+// and breakpoints
+//
+// Breakpoints:
+// sm: < 481
+// md: >= 481
+// lg: >= 769px
+// xl: >= 1025px
+// xxl: >= 1201px
+//
+// Eg. [ 10, 15, 20, 25, 30 ] means { sm: 10, md: 15, lg: 20, xl: 25, xxl: 30 }
+// [10] and [10, 15] are equivalent to { sm: 10 } and { sm: 10, md: 15 } respectively
+//
+// More Info: https://discourse.joplinapp.org/t/week-4-report/26117
+import { Dimensions } from 'react-native';
+
+interface ValueMap {
+    sm?: number;
+    md?: number;
+    lg?: number;
+    xl?: number;
+    xxl?: number;
+}
+
+export default function getResponsiveValue(valueMap: number[] | ValueMap): number {
+	const width = Dimensions.get('screen').width;
+	let value: number;
+	let sm: number, md: number, lg: number, xl: number, xxl: number;
+
+	if (Array.isArray(valueMap)) {
+		if (valueMap.length === 0) {
+			return undefined;
+		}
+
+		[sm, md, lg, xl, xxl] = valueMap;
+
+		value = sm;
+	} else if (Object.keys(valueMap).length === 0) {
```

Comment:
Why would the caller provide an empty value map? Shouldn't you throw an exception in this case?

Reply:
There's no reason for an empty value map, I'll fix that.